chia/full_node/mempool.py

```diff
@@ -8,31 +8,39 @@
 
 
 class Mempool:
-    spends: Dict[bytes32, MempoolItem]
-    sorted_spends: SortedDict  # Dict[float, Dict[bytes32, MempoolItem]]
-    additions: Dict[bytes32, MempoolItem]
-    removals: Dict[bytes32, MempoolItem]
-    size: int
-
-    # if new min fee is added
-    @staticmethod
-    def create(size: int):
-        self = Mempool()
-        self.spends = {}
-        self.additions = {}
-        self.removals = {}
-        self.sorted_spends = SortedDict()
-        self.size = size
-        return self
-
-    def get_min_fee_rate(self) -> float:
-        if self.at_full_capacity():
-            fee_per_cost, val = self.sorted_spends.peekitem(index=0)
-            return fee_per_cost
+    def __init__(self, max_size_in_cost: int):
+        self.spends: Dict[bytes32, MempoolItem] = {}
+        self.sorted_spends: SortedDict = SortedDict()
+        self.additions: Dict[bytes32, MempoolItem] = {}
+        self.removals: Dict[bytes32, MempoolItem] = {}
+        self.max_size_in_cost: int = max_size_in_cost
+        self.total_mempool_cost: int = 0
+
+    def get_min_fee_rate(self, cost: int) -> float:
+        """
+        Gets the minimum fpc rate that a transaction with specified cost will need in order to get included.
+        """
+
+        if self.at_full_capacity(cost):
+            current_cost = self.total_mempool_cost
+
+            # Iterates through all spends in increasing fee per cost
+            for fee_per_cost, spends_with_fpc in self.sorted_spends.items():
+                for spend_name, item in spends_with_fpc.items():
+                    current_cost -= item.cost_result.cost
+                    # Removing one at a time, until our transaction of size cost fits
+                    if current_cost + cost <= self.max_size_in_cost:
+                        return fee_per_cost
+            raise ValueError(
+                f"Transaction with cost {cost} does not fit in mempool of max cost {self.max_size_in_cost}"
+            )
         else:
             return 0
 
-    def remove_spend(self, item: MempoolItem):
+    def remove_from_pool(self, item: MempoolItem):
```

Comment:
Maybe `eject` is more accurately descriptive (since `self` is the pool)?

Reply:
I guess `remove` is accurate too... `eject` emphasizes the "get out of here" aspect.